# Introduction

In [1]:
import numpy as numpy
import pandas as pd
import re


# Examining the data frame 

In [2]:
perscore_df = pd.read_csv("personality_scores.csv" , delimiter = ";")
perscore_df.head(5)

ID Section 5 of 6 [I am always prepared.]  \
0   0                                 (3, 5)   
1   1                                 (3, 5)   
2   2                                 (3, 5)   
3   3                                 (3, 5)   
4   4                                 (3, 3)   

  Section 5 of 6 [I am easily disturbed.]  \
0                                  (4, 5)   
1                                  (4, 5)   
2                                  (4, 3)   
3                                  (4, 5)   
4                                  (4, 5)   

  Section 5 of 6 [I am exacting (demanding) in my work.]  \
0                                             (3, 5)       
1                                             (3, 5)       
2                                             (3, 3)       
3                                             (3, 3)       
4                                             (3, 3)       

  Section 5 of 6 [I am full of ideas.]  \
0                               (5, 5)   
1                               (5, 5)   
2                               (5, 5)   
3                               (5, 5)   
4                               (5, 3)   

  Section 5 of 6 [I am interested in people.]  \
0                                      (2, 3)   
1                                      (2, 5)   
2                                      (2, 5)   
3                                      (2, 5)   
4                                      (2, 3)   

  Section 5 of 6 [I am not interested in abstract ideas.]  \
0                                             (5, 3)        
1                                             (5, 3)        
2                                             (5, 5)        
3                                             (5, 3)        
4                                             (5, 3)        

  Section 5 of 6 [I am not interested in other people's problems.]  \
0                                             (2, 3)                 
1                                             (2, 5)                 
2                                             (2, 5)                 
3                                             (2, 3)                 
4                                             (2, 3)                 

  Section 5 of 6 [I am not really interested in others.]  \
0                                             (2, 5)       
1                                             (2, 5)       
2                                             (2, 5)       
3                                             (2, 3)       
4                                             (2, 3)       

  Section 5 of 6 [I am quick to understand things.]  ... Unnamed: 60  \
0                                            (5, 5)  ...         NaN   
1                                            (5, 5)  ...         NaN   
2                                            (5, 5)  ...         NaN   
3                                            (5, 3)  ...         NaN   
4                                            (5, 5)  ...         NaN   

  Unnamed: 61 Unnamed: 62 Unnamed: 63 Unnamed: 64 Unnamed: 65 Unnamed: 66  \
0         NaN         NaN         NaN         NaN         NaN         NaN   
1         NaN         NaN         NaN         NaN         NaN         NaN   
2         NaN         NaN         NaN         NaN         NaN         NaN   
3         NaN         NaN         NaN         NaN         NaN         NaN   
4         NaN         NaN         NaN         NaN         NaN         NaN   

  Unnamed: 67 Unnamed: 68 IPIP_HIGH_RISK  
0         NaN         NaN            NaN  
1         NaN         NaN            NaN  
2         NaN         NaN            NaN  
3         NaN         NaN            NaN  
4         NaN         NaN            NaN  

[5 rows x 70 columns]

# Checking Missing values

In [3]:
perscore_df.isnull().sum()


ID                                                           0
Section 5 of 6 [I am always prepared.]                       0
Section 5 of 6 [I am easily disturbed.]                      0
Section 5 of 6 [I am exacting (demanding) in my work.]       0
Section 5 of 6 [I am full of ideas.]                         0
                                                          ... 
Unnamed: 65                                               1555
Unnamed: 66                                               1555
Unnamed: 67                                               1555
Unnamed: 68                                               1555
IPIP_HIGH_RISK                                            1555
Length: 70, dtype: int64

# Checking for duplicates

In [4]:
#Checking the number of unique values
perscore_unique = perscore_df["ID"].nunique()
perscore_unique


1555

There are 1555 unique values in the ID meaning that there are no duplicates in the ID of the dataframe

# Removing duplicates

In [5]:
for score in perscore_df:
    if "ID" in perscore_df:
        new_df = perscore_df.drop_duplicates(subset = ["ID"], keep = "first", inplace = False)

# Dropping unnecessary columns

In [6]:
#Dropping Unnamed: column from data frame because it has null values
new_df.dropna(axis = 1,inplace = True)
new_df.head(5)

ID Section 5 of 6 [I am always prepared.]  \
0   0                                 (3, 5)   
1   1                                 (3, 5)   
2   2                                 (3, 5)   
3   3                                 (3, 5)   
4   4                                 (3, 3)   

  Section 5 of 6 [I am easily disturbed.]  \
0                                  (4, 5)   
1                                  (4, 5)   
2                                  (4, 3)   
3                                  (4, 5)   
4                                  (4, 5)   

  Section 5 of 6 [I am exacting (demanding) in my work.]  \
0                                             (3, 5)       
1                                             (3, 5)       
2                                             (3, 3)       
3                                             (3, 3)       
4                                             (3, 3)       

  Section 5 of 6 [I am full of ideas.]  \
0                               (5, 5)   
1                               (5, 5)   
2                               (5, 5)   
3                               (5, 5)   
4                               (5, 3)   

  Section 5 of 6 [I am interested in people.]  \
0                                      (2, 3)   
1                                      (2, 5)   
2                                      (2, 5)   
3                                      (2, 5)   
4                                      (2, 3)   

  Section 5 of 6 [I am not interested in abstract ideas.]  \
0                                             (5, 3)        
1                                             (5, 3)        
2                                             (5, 5)        
3                                             (5, 3)        
4                                             (5, 3)        

  Section 5 of 6 [I am not interested in other people's problems.]  \
0                                             (2, 3)                 
1                                             (2, 5)                 
2                                             (2, 5)                 
3                                             (2, 3)                 
4                                             (2, 3)                 

  Section 5 of 6 [I am not really interested in others.]  \
0                                             (2, 5)       
1                                             (2, 5)       
2                                             (2, 5)       
3                                             (2, 3)       
4                                             (2, 3)       

  Section 5 of 6 [I am quick to understand things.]  ...  \
0                                            (5, 5)  ...   
1                                            (5, 5)  ...   
2                                            (5, 5)  ...   
3                                            (5, 3)  ...   
4                                            (5, 5)  ...   

  Section 5 of 6 [I often forget to put things back in their proper place]  \
0                                             (3, 5)                         
1                                             (3, 5)                         
2                                             (3, 5)                         
3                                             (3, 1)                         
4                                             (3, 5)                         

  Section 5 of 6 [I pay attention to details.]  \
0                                       (3, 5)   
1                                       (3, 1)   
2                                       (3, 5)   
3                                       (3, 5)   
4                                       (3, 5)   

  Section 5 of 6 [I seldom feel blue (down).]  \
0                                      (4, 3)   
1                                      (4, 1)   
2                                      (4, 1)   
3                                      (4, 1)   
4        

# Renaming my dataframe to remove regular expressions

In [7]:
new_df.rename(columns = {"Section 5 of 6 [I am always prepared.]": "I am always prepared"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I am easily disturbed.]": "I am easily disturbed"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I am exacting (demanding) in my work.]": "I am exacting (demanding) in my work"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I am full of ideas.]": "I am full of ideas"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I am interested in people.]": "I am interested in people"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I am always prepared.]": "I am always prepared"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I am easily disturbed.]": "I am easily disturbed"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I am exacting (demanding) in my work.]": "I am exacting (demanding) in my work"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I am full of ideas.]": "I am full of ideas"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I am interested in people.]": "I am interested in people"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I am not interested in abstract ideas.]": "I am not interested in abstract ideas."}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I am not interested in other people's problems.]": "I am not interested in other people's problems"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I am not really interested in others.]": "I am not really interested in others"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I am quick to understand things.]": "I am quick to understand things"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I am quiet around strangers.]": "I am quiet around strangers"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I pay attention to details.]": "I pay attention to details"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I seldom feel blue (down).]": "I seldom feel blue (down)"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I spend time reflecting on things.]": "I spend time reflecting on things"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I start conversations.]": "I start conversations"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I sympathize with others' feelings.]": "I sympathize with others' feelings"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I take time out for others.]": "I take time out for others"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I talk to a lot of different people at parties.]": "I talk to a lot of different people at parties"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I use difficult words.]": "I use difficult words"}, inplace = True)
new_df.rename(columns = {"Section 5 of 6 [I worry about things.]": "I worry about things"}, inplace = True)

new_df.head(5)


ID I am always prepared I am easily disturbed  \
0   0               (3, 5)                (4, 5)   
1   1               (3, 5)                (4, 5)   
2   2               (3, 5)                (4, 3)   
3   3               (3, 5)                (4, 5)   
4   4               (3, 3)                (4, 5)   

  I am exacting (demanding) in my work I am full of ideas  \
0                               (3, 5)             (5, 5)   
1                               (3, 5)             (5, 5)   
2                               (3, 3)             (5, 5)   
3                               (3, 3)             (5, 5)   
4                               (3, 3)             (5, 3)   

  I am interested in people I am not interested in abstract ideas.  \
0                    (2, 3)                                 (5, 3)   
1                    (2, 5)                                 (5, 3)   
2                    (2, 5)                                 (5, 5)   
3                    (2, 5)                                 (5, 3)   
4                    (2, 3)                                 (5, 3)   

  I am not interested in other people's problems  \
0                                         (2, 3)   
1                                         (2, 5)   
2                                         (2, 5)   
3                                         (2, 3)   
4                                         (2, 3)   

  I am not really interested in others I am quick to understand things  ...  \
0                               (2, 5)                          (5, 5)  ...   
1                               (2, 5)                          (5, 5)  ...   
2                               (2, 5)                          (5, 5)  ...   
3                               (2, 3)                          (5, 3)  ...   
4                               (2, 3)                          (5, 5)  ...   

  Section 5 of 6 [I often forget to put things back in their proper place]  \
0                                             (3, 5)                         
1                                             (3, 5)                         
2                                             (3, 5)                         
3                                             (3, 1)                         
4                                             (3, 5)                         

  I pay attention to details I seldom feel blue (down)  \
0                     (3, 5)                    (4, 3)   
1                     (3, 1)                    (4, 1)   
2                     (3, 5)                    (4, 1)   
3                     (3, 5)                    (4, 1)   
4                     (3, 5)                    (4, 5)   

  I spend time reflecting on things I start conversations  \
0                            (5, 5)                (1, 3)   
1                            (5, 5)                (1, 5)   
2                            (5, 3)                (1, 3)   
3                            (5, 5)                (1, 5)   
4                            (5, 5)                (1, 3)   

  I sympathize with others' feelings I take time out for others  \
0                             (2, 5)                     (2, 5)   
1                             (2, 5)                     (2, 5)   
2                             (2, 5)                     (2, 5)   
3                             (2, 5)                     (2, 5)   
4                             (2, 3)                     (2, 5)   

  I talk to a lot of different people at parties I use difficult words  \
0                                         (1, 3)                (5, 1)   
1                                         (1, 5)                (5, 3)   
2                                         (1, 3)                (5, 1)   
3                                         (1, 5)                (5, 1)   
4                                         (1, 3)                (5, 1)   

  I worry about things  
0               (4, 3)  
1               (4, 3)  
2     

# Setting the ID column to be the index of the dataframe

In [8]:
per_index = new_df.set_index("ID")
per_index.head(5)

I am always prepared I am easily disturbed  \
ID                                              
0                (3, 5)                (4, 5)   
1                (3, 5)                (4, 5)   
2                (3, 5)                (4, 3)   
3                (3, 5)                (4, 5)   
4                (3, 3)                (4, 5)   

   I am exacting (demanding) in my work I am full of ideas  \
ID                                                           
0                                (3, 5)             (5, 5)   
1                                (3, 5)             (5, 5)   
2                                (3, 3)             (5, 5)   
3                                (3, 3)             (5, 5)   
4                                (3, 3)             (5, 3)   

   I am interested in people I am not interested in abstract ideas.  \
ID                                                                    
0                     (2, 3)                                 (5, 3)   
1                     (2, 5)                                 (5, 3)   
2                     (2, 5)                                 (5, 5)   
3                     (2, 5)                                 (5, 3)   
4                     (2, 3)                                 (5, 3)   

   I am not interested in other people's problems  \
ID                                                  
0                                          (2, 3)   
1                                          (2, 5)   
2                                          (2, 5)   
3                                          (2, 3)   
4                                          (2, 3)   

   I am not really interested in others I am quick to understand things  \
ID                                                                        
0                                (2, 5)                          (5, 5)   
1                                (2, 5)                          (5, 5)   
2                                (2, 5)                          (5, 5)   
3                                (2, 3)                          (5, 3)   
4                                (2, 3)                          (5, 5)   

   I am quiet around strangers  ...  \
ID                              ...   
0                       (1, 3)  ...   
1                       (1, 3)  ...   
2                       (1, 1)  ...   
3                       (1, 3)  ...   
4                       (1, 1)  ...   

   Section 5 of 6 [I often forget to put things back in their proper place]  \
ID                                                                            
0                                              (3, 5)                         
1                                              (3, 5)                         
2                                              (3, 5)                         
3                                              (3, 1)                         
4                                              (3, 5)                         

   I pay attention to details I seldom feel blue (down)  \
ID                                                        
0                      (3, 5)                    (4, 3)   
1                      (3, 1)                    (4, 1)   
2                      (3, 5)                    (4, 1)   
3                      (3, 5)                    (4, 1)   
4                      (3, 5)                    (4, 5)   

   I spend time reflecting on things I start conversations  \
ID                                                           
0                             (5, 5)                (1, 3)   
1                             (5, 5)                (1, 5)   
2                             (5, 3)                (1, 3)   
3                             (5, 5)                (1, 5)   
4                             (5, 5)                (1, 3)   

   I sympathize with others' feelings I take time out for others  \
ID                                                                 
0              

# Checking is new data frame == original dataframe and asserting new data frame with original dataframe

In [9]:
if  len(perscore_df) == len(new_df):
    print(True)
else:
    print(False)

True


In [10]:
assert len(perscore_df) == len(new_df)

# Calculating total score